### PR TITLE
18/05 cache-decorator

### PR DESCRIPTION
**Pull request by NikitaNikolaitsev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+
+    def inner(*args: Any) -> Any:
+        if args in cache_dict:
+            print("Getting from cache")
+            return cache_dict[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_dict[args] = result
+            return result
```
Comment (DevSpaciX) [inline app/main.py]: Think how you can call return only once . If `cache_dict[args] = result ` you can return  cache_dict[args] and it is makes 2 identical returns in your func . Optimise it : )

**Review by DevSpaciX — CHANGES_REQUESTED** [review]

**Review by wQuelS — CHANGES_REQUESTED** [review]
all tests should pass, including flake8 routine

**Review by wQuelS — CHANGES_REQUESTED** [review]
reject, since not all GH actions are green

**Review by wQuelS — APPROVED** [review]
Looks good to me

**Pull request closed without merging**